### ISSUE TITLE
prov/opx: Build failure in the absence of libnuma

### ISSUE THREAD

**Issue by rajachan** [ofiwg/libfabric#7950]
**Describe the bug**

When a provider is not explicitly requested at configure time, we build any provider that can be built based on the dependencies available on the build system. OPX seems to fail configuration when libnuma is not present even with --enable-opx is not explicitly set.  One would have to explicitly disable OPX or build with `--enable-only` on such a system for the configuration to complete.

As reported in: https://github.com/spack/spack/pull/32083

**To Reproduce**

@mpokorny, please add your exact build flags here.

**Expected behavior**

When --enable-opx is not set, the OPX provider should disable itself when all its dependencies are not satisfied.

**Output**

```
623    libtool: compile:  /home/martin/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. 
            -I./include -D_GNU_SOURCE -D__USE_XOPEN2K8 -DSYSCONFDIR=\"/home/martin/spack/opt
            /spack/linux-pop22-skylake/gcc-11.2.0/libfabric-1.15.1-bfoc34edy5ikfkuwyskjg7drw
            2uamghb/etc\" -DRDMADIR=\"@rdmadir@\" -DPROVDLDIR=\"/home/martin/spack/opt/spack
            /linux-pop22-skylake/gcc-11.2.0/libfabric-1.15.1-bfoc34edy5ikfkuwyskjg7drw2uamgh
            b/lib/libfabric\" -I./prov/sockets/include -I./prov/sockets -I./prov/opx/include
             -I./prov/opx/include -I./prov/opx/include/linux-i386 -I./prov/hook/include -I./
            prov/hook/perf/include -I./prov/hook/hook_debug/include -I./prov/hook/hook_hmem/
            include -I./prov/hook/dmabuf_peer_mem/include -Wall -O2 -DNDEBUG -fvisibility=hi
            dden -Wall -Wundef -Wpointer-arith -MT prov/opx/src/src_libfabric_la-fi_opx_hfi1
            .lo -MD -MP -MF prov/opx/src/.deps/src_libfabric_la-fi_opx_hfi1.Tpo -c prov/opx/
            src/fi_opx_hfi1.c  -fPIC -DPIC -o prov/opx/src/.libs/src_libfabric_la-fi_opx_hfi
            1.o
  >> 624    prov/opx/src/fi_opx_hfi1.c:34:10: fatal error: numa.h: No such file or directory
     625       34 | #include <numa.h>
     626          |          ^~~~~~~~
     627    compilation terminated.
  >> 628    make[1]: *** [Makefile:22008: prov/opx/src/src_libfabric_la-fi_opx_hfi1.lo] Erro
            r 1
```

**Comment by shefty** [comment]
Is this with the latest main or v1.15.2rc2?  There were fixes applied to the opx provider build I believe to address this.  See e40039839fe88a432f1e42ec7c3d7628a0439bdf

**Comment by mpokorny** [comment]
It was with v1.15.1 (in updated Spack package)

**Comment by mpokorny** [comment]
Configure flags: `--disable-debug --with-kdreg=no --enable-efa=no --enable-gni=no --enable-mlx=no --enable-mrail=no --enable-psm=no --enable-psm2=no --enable-psm3=no --enable-rxm=no --enable-rxd=no --enable-shm=no --enable-sockets=yes --enable-tcp=yes --enable-udp=yes --enable-usnic=no --enable-verbs=no --enable-xpmem=no` (note that Spack package knows nothing about OPX yet).
Compile flags are shown in @rajachan 's initial message.

**Comment by shefty** [comment]
I believe this is now fixed and will be in v1.15.2.  The commit I referenced above is in rc2.

**Comment by shefty** [comment]
Btw, the rc2 commit is e3906cab80315636cf929c6d904a8689e5daffae (changes were needed to resolve a conflict because patches were merged into v1.15.x out of order from main0.
